### ISSUE TITLE
RFE: add `--debug` to `build` and `--release` to `install`

### ISSUE THREAD

**Issue by ensc** [rust-lang/cargo#7720]
With traditional build systems I am using a workflow like

```
make build CFLAGS='-O2 -g3'
make check
make install DESTDIR=`pwd`/i
```

to 

1. build the software

2. run tests

3. install the built artifacts

With cargo, this is more complicated because `cargo install` executes the `build` step and ignores the previous `cargo build`.  But `build` and `install` expect different options; e..g. you have to do either

```
cargo build             --feature foo
cargo install --debug   --feature foo --path . --root `pwd`/i 
```

or

```
cargo build   --release --feature foo
cargo install           --feature foo --path . --root `pwd`/i 
```

This require to carry around two different options expressing the same.

Things would be more easy when both `--debug` and `--release` are available to `build` + `install`.  Then, you can just call

```
cargo build   $CARGO_FLAGS
cargo install $CARGO_FLAGS --path . --root `pwd`/i 
```

and use a common `CARGO_FLAGS` variable.

**Comment by ehuss** [comment]
I think we're moving towards using `--profile=release` and `--profile=debug` instead of adding more `--release` and `--debug` flags.  It is implemented on nightly, and is tracked in #6988.

**Comment by ehuss** [comment]
Closing since we will likely be using the `--profile` flag.
